src/server/detail/decompress.h

```diff
@@ -0,0 +1,30 @@
+// Copyright 2024, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+#pragma once
+
+#include "io/io.h"
+#include "io/io_buf.h"
+
+namespace dfly {
+
+namespace detail {
```

Comment:
I'd even move this to `core/`

Reply:
we could, but I think at this time it does not really matter.